### PR TITLE
[node] Add missing alpnProtocol and createConnection

### PR DESCRIPTION
**Pull request by tlbdk**
Please fill in this template.

- [ x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x ] Test the change in your own code. (Compile and run.)
- [ x ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ x ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ x ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ x ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ x ] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/tls.html#tls_event_secureconnection , https://nodejs.org/api/http2.html#http2_http2_connect_authority_options_listener

### CODE REVIEW

File: types/node/index.d.ts
Comment (isLishude) [inline types/node/index.d.ts]: I don't find the `alpnProtocol` here.Are you sure this needs it?

**Review by ZaneHannanAU — APPROVED** [review]

File: types/node/index.d.ts
Comment (ZaneHannanAU) [reply types/node/index.d.ts]: @isLishude it's used in the http2 documents for when checking whether to use push or not.

https://nodejs.org/api/http2.html#http2_alpn_negotiation

Comment (SimonSchick) [reply types/node/index.d.ts]: This should be `alpnProtocol?: string` then?

File: types/node/index.d.ts
Comment (SimonSchick) [inline types/node/index.d.ts]: Make this `createConnection?(...): stream.Duplex`

**Review by SimonSchick — CHANGES_REQUESTED** [review]

File: types/node/index.d.ts
Comment (tlbdk) [reply types/node/index.d.ts]: Not quite sure I understand, the function takes a callback that gets the SessionOptions as args and returns a stream.Duplex ?

File: types/node/index.d.ts
Comment (tlbdk) [reply types/node/index.d.ts]: Fixed

**Review by BrunoScheufler — APPROVED** [review]

**Comment by tlbdk** [comment]
@SimonSchick anything more to fix?

**Comment by tlbdk** [comment]
Is there any way to only build the node parts in the CI, it seems a bit overkill to build everything

**Comment by Flarna** [comment]
@tlbdk CI has to build node + all dependents (>1000) otherwise it would be too easy to break one of them.
I think it's possible to do this local, at least the typescript part.

**Review by SimonSchick — APPROVED** [review]

**Comment by tlbdk** [comment]
@Flarna Does the build need to pass before this can be merged? 

Because it seems something unrelated is breaking for the pull request

**Comment by Flarna** [comment]
@tlbdk To my understanding the repo maintainers have only PRs on their radar which have the `Merge:Express` label on it. To my knowledge typescript-bot adds this only if CI is green and the PR is approved.

In the current case CI fail is because of timeout. This may be because of some temporary overload in Travis so it's worth to try a re-trigger.

If retry doesn't work, or the fail is because of issues in other modules I tend to ping the repo maintainers (see e.g. https://github.com/DefinitelyTyped/DefinitelyTyped/pull/30346#issuecomment-437379362) and ask for merge even CI has failed.

**Comment by tlbdk** [comment]
@Flarna Thanks, it also seems there is some logic on what tests are run based on a simple diff with master, I merged with master on last build so that list was rather long, now I have rebased in on the latest master, hope this helps.

**Comment by Flarna** [comment]
Well the changed modules + all modules depending on them are tests. As you change node which is used by 1941 (extracted from current CI run) it's a lot stuff which is tested.
And if any of this packages has an issue - caused by your changes or not - CI fails.

**Comment by SimonSchick** [comment]
@andy-ms can we perhaps add special behaviour for the typescript-bot to display additional information if  a CI build for a large package (such as node) failed?
I think it's confusing a lot of people at this point.

**Comment by Flarna** [comment]
@andy-ms @RyanCavanaugh Seems the labels on this ticket are wrong. It still shows "Revison needed" even the comments have been addressed and approved.

Additionally CI fail is unrelated.

Any chance to get this one merged?

**Comment by tlbdk** [comment]
Do I need to do anything more on this?

**Pull request closed without merging**
